packages/next/src/builders/build/lib/create-package-json.ts

```diff
@@ -2,13 +2,41 @@ import { BuilderContext } from '@angular-devkit/architect';
 import { writeFileSync } from 'fs';
 import { join } from 'path';
 import { readJsonFile } from '@nrwl/workspace';
+import { createProjectGraph } from '@nrwl/workspace/src/core/project-graph';
 import { NextBuildBuilderOptions } from '../../../utils/types';
 import { reactDomVersion, reactVersion } from '@nrwl/react/src/utils/versions';
 
+interface PackageJSONDeps {
+  dependencies: { [key: string]: string };
+  devDependencies: { [key: string]: string };
+  peerDependencies?: { [key: string]: string };
+  [key: string]: unknown;
+}
+
 export async function createPackageJson(
   options: NextBuildBuilderOptions,
   context: BuilderContext
 ) {
+  const graph = createProjectGraph();
+  const projectDependencies = graph.dependencies[context.target.project].map(
+    (dep) => dep.target
+  );
+  const projectNpmDependencies = Object.values(graph.nodes).reduce(
```

Comment:
I think it's simpler and more performant to keep mapping the above array to get to the npm projects?

```ts
const projectDependencies = graph.dependencies[context.target.project]
  .map(dep => dep.target)
  .map(nodeName => graph.nodes[nodeName])
  .filter(node => node.type === 'npm')
  .reduce(...)
```